returnn/config.py

```diff
@@ -131,6 +131,40 @@ def get(self, key, default=None):
             return self.post_config[key]
         return self.config.get(key, default)
 
+    def update(self, other):
+        """
+        updates a ReturnnConfig with an other ReturnnConfig
+
+        :param ReturnnConfig other:
+        """
+        self.config.update(other.config)
+        self.post_config.update(other.post_config)
+        if other.staged_network_dict is not None:
+            self.staged_network_dict = other.staged_network_dict
```

Comment:
I would prefer also `update` here instead of overwriting, this would be more consistent.

Reply:
I specifically decided for overwriting the `staged_network_dict` instead of updating with the use-case of "entirely replacing the network architecture" in mind.

Assume you have `baseline_conformer_training_config` and want to do experiments with different layer dimension, or transformer, or linformer,.. then you might use 
```
baseline_conformer_training_config.update(my_network_config)
```
In the update case, if now the stage epoch numbers do not exactly coincide it is easy to accidentally mess up the pretraining and hard to intentionally design `my_network_config` to do it correctly.

In the overwrite case, if you expected "update behavior" you would most likely immediately notice your error as the epoch=0 stage would be missing and if you intentionally want to do the correct thing, then 
```
baseline_conformer_training_config.staged_network_dict.update(my_stage_config.staged_network_dict)
```
is only minimal overhead.

Reply:
Ok, I take your argument as valid :) Then please keep the full replacement.